pkg/utils/integration/snapshots.go

```diff
@@ -0,0 +1,136 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/devfile/library/v2/pkg/util"
+	appstudioApi "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateSnapshotWithImage creates a snapshot using an image.
+func (i *IntegrationController) CreateSnapshotWithImage(applicationName, namespace, componentName, containerImage string) (*appstudioApi.Snapshot, error) {
+	hasSnapshot := &appstudioApi.Snapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "snapshot-sample-" + util.GenerateRandomString(4),
+			Namespace: namespace,
+			Labels: map[string]string{
+				"test.appstudio.openshift.io/type":           "component",
+				"appstudio.openshift.io/component":           componentName,
+				"pac.test.appstudio.openshift.io/event-type": "push",
+			},
+		},
+		Spec: appstudioApi.SnapshotSpec{
+			Application: applicationName,
+			Components: []appstudioApi.SnapshotComponent{
+				{
+					Name:           componentName,
+					ContainerImage: containerImage,
+				},
+			},
+		},
+	}
+	err := i.KubeRest().Create(context.TODO(), hasSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	return hasSnapshot, err
+}
+
+// CreateSnapshotWithComponents creates a Snapshot using the given parameters.
+func (i *IntegrationController) CreateSnapshotWithComponents(applicationName, namespace, snapshotName string, snapshotComponents []appstudioApi.SnapshotComponent) (*appstudioApi.Snapshot, error) {
```

Comment:
Curious, any specific reason why you swapped the order of the 2 parameters (`applicationName` and `snapshotName`)?

I think it made sense to have the first parameter as `snapshotName` because the main aim of this method is to create a Snapshot.

Reply:
I swapped them to make order of parameters in CreateSnapshotWithImage() and CreateSnapshotWithComponents() unified.  Based on this [convo](https://github.com/redhat-appstudio/e2e-tests/pull/644#discussion_r1273417394). Wdyt keep or revert?

Reply:
Oops, sorry about that. I added my comment there.